### PR TITLE
bugfix in ptpconfig discovery,  fix for logging and additional logs

### PR DESCRIPTION
**Pull request by edcdavid**
Some bugfxes in ptpconfig discovery, logging, adding additional logs
also includes https://github.com/openshift/ptp-operator/pull/227 and skipping test old test waiting for clock ID when in discovery mode

### CODE REVIEW

File: test/utils/metrics/metrics.go [hunk 1]
```diff
@@ -131,6 +131,12 @@ func CheckClockRoleAndOffset(ptpConfig *ptpv1.PtpConfig, label, nodeName *string
 	if nodeName == nil {
 		var name string
 		name, err = getNode(*label)
+		if err != nil || name == "" || label == nil || *label != utils.PtpClockUnderTestNodeLabel {
+			fmt.Printf(`error getting node name for label %s
```
Comment (aneeshkp) [inline test/utils/metrics/metrics.go:135]: log instead of fmt

File: test/conformance/ptp/ptp.go [hunk 2]
```diff
@@ -141,6 +141,12 @@ var _ = Describe("[ptp]", func() {
 		execute.BeforeAll(func() {
 			testconfig.CreatePtpConfigurations()
 			fullConfig = testconfig.GetFullDiscoveredConfig(utils.PtpLinuxDaemonNamespace, false)
+			if fullConfig.Status != testconfig.DiscoverySuccessStatus {
+				fmt.Printf(`ptpconfigs were not properly discovered, Check:
```
Comment (aneeshkp) [inline test/conformance/ptp/ptp.go:145]: log instead of fmt.

File: test/utils/metrics/metrics.go [hunk 3]
```diff
@@ -90,9 +90,9 @@ func getMetric(nodeName, aIf, metricName string) (metric string, err error) {
 		var regex string
 		if metricName == OpenshiftPtpOffsetNs {
 			aIf = aIf[:len(aIf)-1] + "x"
-			regex = metricName + `{` + fromMaster + `iface="` + aIf + `",node="` + ptpPods.Items[index].Spec.NodeName + `",process="ptp4l"} ([0-9]*)`
+			regex = metricName + `{` + fromMaster + `iface="` + aIf + `",node="` + ptpPods.Items[index].Spec.NodeName + `",process="ptp4l"} (-*[0-9]*)`
 		} else {
-			regex = metricName + `{iface="` + aIf + `",node="` + ptpPods.Items[index].Spec.NodeName + `",process="ptp4l"} ([0-9]*)`
+			regex = metricName + `{iface="` + aIf + `",node="` + ptpPods.Items[index].Spec.NodeName + `",process="ptp4l"} (-*[0-9]*)`
```
Comment (aneeshkp) [inline test/utils/metrics/metrics.go:95]: This should not be the case I think . There should be an metrics without from

File: test/utils/metrics/metrics.go [hunk 4]
```diff
@@ -90,9 +90,9 @@ func getMetric(nodeName, aIf, metricName string) (metric string, err error) {
 		var regex string
 		if metricName == OpenshiftPtpOffsetNs {
 			aIf = aIf[:len(aIf)-1] + "x"
```
Comment (aneeshkp) [inline test/utils/metrics/metrics.go:92]: could you use  regex for aIf, since its not significant

File: test/utils/metrics/metrics.go [hunk 5]
```diff
@@ -131,6 +131,12 @@ func CheckClockRoleAndOffset(ptpConfig *ptpv1.PtpConfig, label, nodeName *string
 	if nodeName == nil {
 		var name string
 		name, err = getNode(*label)
+		if err != nil || name == "" || label == nil || *label != utils.PtpClockUnderTestNodeLabel {
```
Comment (aneeshkp) [inline test/utils/metrics/metrics.go:134]: what would be the label incase of multi cluster for master(leader) and slave(follower) ? do you need to have that label defined

File: test/utils/metrics/metrics.go (hunk 1 above)
Comment (edcdavid) [reply test/utils/metrics/metrics.go:135]: This is the last log of the output, using fmt for better formatting

File: test/utils/metrics/metrics.go (hunk 3 above)
Comment (edcdavid) [reply test/utils/metrics/metrics.go:95]: This handles several metric types

**Pull request merged**